friByte.capture-the-flag.service/friByte.capture-the-flag.service/Services/CtfTaskService.cs

```diff
@@ -196,13 +196,37 @@ private async Task SendSolvedTaskEventToClients(SolvedTask solvedTask)
             return;
         }
 
+        bool isFirstBlood = await IsFirstBlood(solvedTask.TaskId, solvedTask.TeamId);
+
         // Runs both calls in parallel
         await Task.WhenAll(
             _ctfSignalrHubContext.Clients.All.ReceiveSolvedTask(new SolvedTaskReadModel(solvedTask)),
             _ctfSignalrHubContext.Clients.All.ReceiveLeaderboardEntryChange(
-                _ctfLeaderboardService.GetScoreForTeamId(solvedTask.TeamId).Result)
+                _ctfLeaderboardService.GetScoreForTeamId(solvedTask.TeamId).Result),
+            NotifyFirstBlood(solvedTask.TaskId, solvedTask.TeamId, isFirstBlood)
         );
     }
+
+    private async Task<bool> IsFirstBlood(Guid taskId, string teamId)
+    {
+        var solvedTasks = await _ctfContext.SolvedTasks
+            .Where(t => t.TaskId == taskId && t.TeamId != teamId)
+            .ToListAsync();
+
+        return !solvedTasks.Any();
```

Comment:
```suggestion
        var isAlreadySolved = await _ctfContext.SolvedTasks
            .Where(t => t.TaskId == taskId && t.TeamId != teamId)
            .AnyAsync();

        return !isAlreadySolved;
```
Not much different, but a bit more efficient as we just return true or false from database instead of the whole object